logging/zap/grpclogger.go

```diff
@@ -44,3 +48,102 @@ func (l *zapGrpcLogger) Printf(format string, args ...interface{}) {
 func (l *zapGrpcLogger) Println(args ...interface{}) {
 	l.logger.Info(fmt.Sprint(args...))
 }
+
+// ReplaceGrpcLoggerV2 replaces the grpc_log.LoggerV2 with the input zap.Logger
+// This logger adheres to the grpc go environment variables GRPC_GO_LOG_VERBOSITY_LEVEL and GRPC_GO_LOG_SEVERITY_LEVEL.
+func ReplaceGrpcLoggerV2(logger *zap.Logger) {
+	zgl := &zapGrpcLoggerV2{Logger: logger.With(SystemField, zap.Bool("grpc_log", true))}
+	verbosity := os.Getenv("GRPC_GO_LOG_VERBOSITY_LEVEL")
+	if v, err := strconv.Atoi(verbosity); err == nil {
+		zgl.verbosity = v
+	}
+
+	logLevel := os.Getenv("GRPC_GO_LOG_SEVERITY_LEVEL")
+	switch logLevel {
```

Comment:
In regards to the changing of logLevel dynamically, there is an [open  #issue](https://github.com/uber-go/zap/issues/581) requesting an option, e.g. `WithOptions(wrapCoreWithLevel(zap.WarnLevel)) // proposed option` to achieve this. 

If you follow that thread there is an [implementation in juju](https://github.com/juju/zaputil/blob/master/zapctx/zapctx.go#L68) which basically creates a `coreWithLevel`. 

If a similar approach to the juju code was used, then the pass through methods would be less "branchy" e.g. removing all the `if l.severity >= warnLevel`  and simply delegating the level checks to the already in place zap internals.